### PR TITLE
Credentials naming and cleanup

### PR DESCRIPTION
**Pull request by jboeuf**
- Removing service_accounts credentials. These credentials just have
  drawbacks compared to service_account_jwt_access credentials, notably
  in terms for security.
- Renaming Google specific credentials with a Google prefix for C and
  C++. This should be done as well for wrapped languages.

This PR depends on #2947

### CODE REVIEW

**Comment by jboeuf** [comment]
@yang-g you may want to wait for #2947 to be merged to master as the diff will be much more readable (right now the PR also contains the #2947 changes).

File: test/cpp/interop/client_helper.cc [hunk 1]
```diff
@@ -107,14 +97,10 @@ std::shared_ptr<Channel> CreateChannelForTestCase(
   snprintf(host_port, host_port_buf_size, "%s:%d", FLAGS_server_host.c_str(),
            FLAGS_server_port);
 
-  if (test_case == "service_account_creds") {
```
Comment (yang-g) [inline test/cpp/interop/client_helper.cc]: Also remove the places in interop_client.h interop_client.cc and client.cc related to service_account_creds as a test case.

Comment (jboeuf) [inline test/cpp/interop/client_helper.cc]: Done.

Comment (yang-g) [inline test/cpp/interop/client_helper.cc]: https://github.com/grpc/grpc/blob/master/test/cpp/interop/client.cc#L71 and
the following references to service_account_creds also need to be removed :)

Comment (jboeuf) [inline test/cpp/interop/client_helper.cc]: woopsies... Thanks for pointing that out. Should be fixed now.

**Comment by yang-g** [comment]
LGTM

**Pull request merged**